tests/cxx/alias_template.cc

```diff
@@ -33,6 +38,21 @@ AliasChain2<IndirectClass> aliasChain;
 template<class T> using Pointer = T*;
 Pointer<int> intPtr;
 
+// Test the used class being nested deeper in the alias
+template <typename T>
+using AliasNested = FullUseTemplateArg<FullUseTemplateArg<T>>;
+
+// IWYU: IndirectClass needs a declaration
+// IWYU: IndirectClass is...*indirect.h
+AliasNested<IndirectClass> aliasNested;
+
+// Test the used class being nested deeper in the alias
```

Comment:
I'm not sure I understand how this is different from `AliasNested` above.
The template argument is used in an evaluated context, as opposed to the unevaluated `sizeof` -- but both lead to the same IWYU result?
I wonder if the `sizeof` template should be renamed to clarify that, or something. Now it just looks like there's two identical tests after each other.

Reply:
They are roughly identical; I suspect they are redundant.  But I felt like it didn't hurt to have an extra test case.  Certainly it's important to test that both `sizeof` and regular use-as-a-variable count as full uses, and the deeper nesting was also important to test.  There was no particular reason to choose one over the other for the deeper nesting test, so I just did both.

I've renamed the templates for extra clarity and deleted a redundant comment.